package/libpiksi/libpiksi/src/settings_daemon.c

```diff
@@ -0,0 +1,551 @@
+ * to stay in sync with successful updates as reported by settings owners.
+ * @version v1.4
+ * @date 2018-02-23
+ */
+
+#include <sys/stat.h>
+#include <unistd.h>
+#include <errno.h>
+
+#include <libpiksi/sbp_pubsub.h>
+#include <libpiksi/util.h>
+#include <libpiksi/logging.h>
+#include <libsbp/settings.h>
+
+#include <libpiksi/settings_daemon.h>
+
+#include <libsettings/settings.h>
+
+#define PUB_ENDPOINT "ipc:///var/run/sockets/settings_client.sub"
+#define SUB_ENDPOINT "ipc:///var/run/sockets/settings_client.pub"
+
+typedef struct {
+  pk_endpoint_t *cmd_ept;
+  const char *command;
+  handle_command_fn handler;
+} control_command_t;
+
+/**
+ * @brief Settings Context
+ *
+ * This is the main context for managing client interactions with
+ * the settings manager. It implements the client messaging context
+ * as well as the list of types and settings necessary to perform
+ * the registration, watching and callback functionality of the client.
+ */
+struct sd_ctx_s {
+  pk_loop_t *loop;
+  sbp_pubsub_ctx_t *pubsub_ctx;
+  settings_t *settings;
+};
+
+static sd_term_fn sd_term_handler = NULL;
+static sd_child_fn sd_child_handler = NULL;
+
+static void destroy(sd_ctx_t **ctx)
+{
+  free(*ctx);
+  *ctx = NULL;
+}
+
+static int send_wrap(void *ctx, uint16_t msg_type, uint8_t len, uint8_t *payload)
+{
+  sd_ctx_t *sd_ctx = (sd_ctx_t *)ctx;
+  return sbp_tx_send(sbp_pubsub_tx_ctx_get(sd_ctx->pubsub_ctx), msg_type, len, payload);
+}
+
+static int send_from_wrap(void *ctx,
+                          uint16_t msg_type,
+                          uint8_t len,
+                          uint8_t *payload,
+                          uint16_t sbp_sender_id)
+{
+  sd_ctx_t *sd_ctx = (sd_ctx_t *)ctx;
+  return sbp_tx_send_from(sbp_pubsub_tx_ctx_get(sd_ctx->pubsub_ctx),
+                          msg_type,
+                          len,
+                          payload,
+                          sbp_sender_id);
+}
+
+static int wait_wrap(void *ctx, int timeout_ms)
+{
+  sd_ctx_t *sd_ctx = (sd_ctx_t *)ctx;
+
+  return pk_loop_run_simple_with_timeout(sd_ctx->loop, timeout_ms);
+}
+
+static void signal_wrap(void *ctx)
+{
+  sd_ctx_t *sd_ctx = (sd_ctx_t *)ctx;
+  sbp_rx_reader_interrupt(sbp_pubsub_rx_ctx_get(sd_ctx->pubsub_ctx));
+}
+
+static int reg_cb_wrap(void *ctx,
+                       uint16_t msg_type,
+                       sbp_msg_callback_t cb,
+                       void *cb_context,
+                       sbp_msg_callbacks_node_t **node)
+{
+  sd_ctx_t *sd_ctx = (sd_ctx_t *)ctx;
+  sbp_rx_callback_register(sbp_pubsub_rx_ctx_get(sd_ctx->pubsub_ctx),
+                           msg_type,
+                           cb,
+                           cb_context,
+                           node);
+}
+
+static int unreg_cb_wrap(void *ctx, sbp_msg_callbacks_node_t **node)
+{
+  sd_ctx_t *sd_ctx = (sd_ctx_t *)ctx;
+  return sbp_rx_callback_remove(sbp_pubsub_rx_ctx_get(sd_ctx->pubsub_ctx), node);
+}
+
+sd_ctx_t *sd_create(void)
```

Comment:
to avoid the `settings_daemon` -> `sd_*` conversion, why don't we just use `pk_settings_*`